File: dotnet-rest-api/SocialMedia.Api/Models/LikeDto.cs
using System.Text.Json.Serialization;
using Newtonsoft.Json;
using Swashbuckle.AspNetCore.Annotations;

namespace SocialMedia.Api.Models
{
    /// <summary>
    /// Data Transfer Object representing a like on a social media post.
    /// </summary>
    public class LikeDto
    {
        /// <summary>Unique identifier for the like.</summary>
        [JsonProperty("id")]
        public string? Id { get; set; }

        /// <summary>ID of the post being liked.</summary>
        [JsonProperty("postId")]
        public string? PostId { get; set; }

        /// <summary>ID of the user who liked the post.</summary>
        [JsonProperty("authorId")]
        public string? AuthorId { get; set; }

        /// <summary>Timestamp when the like was created (UTC).</summary>
        [JsonProperty("createdAt")]
        public DateTime CreatedAt { get; set; }

        /// <summary>
        /// Indicates if the like is active or has been deleted.
        /// </summary>
        [SwaggerIgnore]
        public bool IsActive { get; set; } = true;
    }
}

File: node-graphql-api/src/dtos/LikeDto.js
// LikeDto.js - Data Transfer Object for a like on a social media post
// Mirrors the .NET LikeDto structure for consistency

class LikeDto {
  /**
   * @param {Object} data
   * @param {string|null} data.id
   * @param {string|null} data.postId
   * @param {string|null} data.authorId
   * @param {string} data.createdAt
   * @param {boolean} data.isActive
   */
  constructor({
    id = null,
    postId = null,
    authorId = null,
    createdAt,
    isActive = true
  }) {
    this.id = id;
    this.postId = postId;
    this.authorId = authorId;
    this.createdAt = createdAt;
    this.isActive = isActive;
  }
}

export default LikeDto;
